apps/btc_family/dash/dash_app.c

```diff
@@ -0,0 +1,141 @@
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "dash_app.h"
+
+#include "btc_helpers.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Check if the purpose index is supported by the Dash app.
+ * @details The current supported/allowed purpose index is 0x8000002C
+ *
+ * @param purpose_index The purpose index to be checked
+ *
+ * @return bool Indicates if the provided purpose index is supported
+ * @retval true The provided purpose index is supported
+ * @retval false The provided purpose index is not supported
+ */
+static bool is_purpose_supported(uint32_t purpose_index);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/**
+ * The Dash fork of Bitcoin only supports legacy addresses. No support for
+ * Segwit and Native-Segwit. Hence, bech32_hrp is empty and segwit,
+ * native-segwit xpub version is set to 0.
+ * Values can be verified from the following references:
+ * https://docs.dash.org/projects/core/en/18.0.0/docs/reference/transactions-address-conversion.html#conversion-process
+ */
+const btc_config_t dash_app = {
+    .coin_type = COIN_DASH,
+    .p2pkh_addr_ver = 0x4c,
+    .p2sh_addr_ver = 0x10,
+    .legacy_xpub_ver = 0x0488b21e,
+    .segwit_xpub_ver = 0x00,
+    .nsegwit_xpub_ver = 0x00,
+    .bech32_hrp = "",
+    .lunit_name = "DASH",
+    .name = "Dash",
+
+    .is_purpose_supported = is_purpose_supported,
+};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+static bool is_purpose_supported(uint32_t purpose_index) {
```

Comment:
There are some test cases for derivation path checks in `derivation_path_tests`, can you update them for `DASH` coin?

Reply:
added